### PR TITLE
Fix 236

### PR DESCRIPTION
**Pull request by tlogan2000**
### Pull Request Checklist:
- [X] This PR addresses an already opened issue (for bug fixes / features)
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)
- [X] HISTORY.rst has been updated

### After merging to master and before closing the branch:
- [ ] bumpversion (minor / major / patch) has been called on `master` branch
- [ ] Tags have been pushed

* **What kind of change does this PR introduce?** 
Bug fix, #236

* **Does this PR introduce a breaking change?** 
no

* **Other information**:
subset_gridpoint() will now use `.sel` method if lon, lat dims are in the dataset. Otherwise it uses old version based on distance 

This PR fixes #236

### CODE REVIEW

File: xclim/subset.py [hunk 1]
```diff
@@ -114,36 +114,48 @@ def subset_gridpoint(da, lon, lat, start_yr=None, end_yr=None):
...
-    dist = dist.reshape(shp_orig)
-
-    iy, ix = np.unravel_index(np.argmin(dist, axis=None), dist.shape)
-    xydims = [x for x in da.dims if "time" not in x]
-
-    args = dict()
-    args[xydims[0]] = iy
-    args[xydims[1]] = ix
-    out = da.isel(**args)
+    # check if trying to subset lon and lat
```
Comment (huard) [inline xclim/subset.py]: This will fail if lat or lon == 0.

File: xclim/subset.py [hunk 2]
```diff
@@ -114,36 +114,48 @@ def subset_gridpoint(da, lon, lat, start_yr=None, end_yr=None):
...
+        if hasattr(da, "lon") and hasattr(da, "lat"):
+            # adjust negative/positive longitudes if necessary
+            if np.all(da.lon > 0) and lon < 0:
+                lon += 360
+            if np.all(da.lon < 0) and lon > 0:
+                lon -= 360
+
+            dims = list(da.dims)
+
+            if "lat" in dims and "lon" in dims:
```
Comment (huard) [inline xclim/subset.py]: Add comment to explain the logic.

File: xclim/subset.py (hunk 1 above)
Comment (huard) [reply xclim/subset.py]: line 118

**Comment by tlogan2000** [comment]
> Does it also fix the issue raised by David C ?

The issue of different dimension orders? If yes, then it should be solved as I no longer create a 2d meshgrid out of 1d lon lat dim values but instead use the `sel` method.  The 'distance' method is therefore only used when lon, lat dims are not present (i.e. rlat and rlon) and is calculated directly on the irrelgular 2d lon, lat values already present in the dataset. 

Just to make sure I added a test for this in test_utils in `test_irregular` where I transpose the data before running the subset

**Comment by tlogan2000** [comment]
test failing ... @huard wait a minute.

**Comment by tlogan2000** [comment]
Test should be working now...

File: xclim/subset.py (hunk 1 above)
Comment (tlogan2000) [reply xclim/subset.py]: now checks to see if lon/lat is `None`

**Pull request merged**